### ISSUE TITLE
Support for creating sketches other than by drawing

### ISSUE THREAD

**Issue by twelch** [seasketch/next#616]
Request for discussion cc @underbluewaters, @avmey, @wmcclin, @petermenzies 

I know the legacy shapefile import hasn't found it's way to next, and there are reasons we don't like it.  But I believe it has real need, and could be tweaked to support some additional use cases we have:
- importing a network (sketch collection) of existing MPA's into seasketch. (This feels essential to Azores, and eventually others as greenfield mpa planning becomes no longer the norm but I'm open to thoughts.).  The workaround of producing a visual map layer and trying to trace it for all the MPA's and attributing them by hand is not great and makes it harder and less accurate than it needs to be in my opinion.
- importing handcrafted test sketches produced by a developer into seasketch.
- importing sketches produced externally such as via QGIS, etc. into seasketch (Bermuda this is being done in legacy like it or not, and we would benefit from being able to add them as multipolygon (because it's over 100 polygons derived from data where we don't need them to be individual sketches)
- importing sketch types into seasketch that aren't supported by the drawing tools but seasketch internals and the reports can/will be able to work with (e.g. multipolygons, polygons with holes not produced by preprocessor).

Requirements:
- Be able to add polygons and multipolygon at least.
- full sketch attributes can be provided in the json payload, that seasketch can't otherwise create.
- support for individual sketches or a whole collection.

Simplest implementation, the UX could be:
- after selecting to create an instance of a sketch class, if a superuser you have access to a relatively hidden input, a textbox or something for pasting geojson (sketch geojson specifically) as an alternative to drawing.
- preprocessor on submit may be optional, need to discuss
- sketch attributes could be provided up-front in the text payload, with validation.

This is a "thar be dragons" type feature that would be for powerusers only.  Limiting access will limit misuse.

**Comment by underbluewaters** [comment]
I agree this will be essential to have, primarily for existing MPAs. This has lead to a lot of grief in the past and it's worth listing out why.

  1. We end up with shapes that cannot be practically edited. This is obviously problematic for a collaborative design process if shapes can't be iterated on. The *Simple Shape* -> *Complex Geometry Product* design pattern needs to be preserved somehow and often that's just impossible for shapefile uploads. (For use-cases like drawing a circle, we could address that with digitizing tools instead of uploads).
  2. People often create shapes that are clipped to the shoreline... a different shoreline. If we perform clipping again that creates even more needlessly complex shapes.
  3. These more complex shapes often can't be analyzed as efficiently and reports may time out. This often hits us and our users unexpectedly, creating an urgent problem to sort out. I've seen 25MB+ sketches uploaded in the past and when that happens there really isn't any practical solution.

Power users and project admins are the ones who usually cause these issues, so limiting these features to just admins wouldn't likely help.

For existing MPAs I think we need to have some sort of workflow that addresses issue 1. In principal everything that is a "sketch" ought to be editable by end users. I'm not sure how to get there though.

**Comment by twelch** [comment]
I would be okay with it only being possible by a superuser, so that you have to work with someone that understand the nuances. of this, and it's only use for use cases that make sense.

**Comment by twelch** [comment]
Another option might be documenting an API we can hit to publish sketches and sketch collections.  Which might have looser requirements than the drawing (e.g. allow for multipolygons)
